samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -16,27 +16,56 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 package org.apache.samza.runtime;
 
+import org.apache.samza.SamzaException;
+import org.apache.samza.execution.ExecutionPlanner;
+import org.apache.samza.execution.ProcessorGraph;
+import org.apache.samza.job.JobRunner;
+import org.apache.samza.operators.StreamGraph;
 import org.apache.samza.operators.StreamGraphBuilder;
 import org.apache.samza.config.Config;
+import org.apache.samza.operators.StreamGraphImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 /**
  * This class implements the {@link ApplicationRunner} that runs the applications in a remote cluster
  */
 public class RemoteApplicationRunner extends AbstractApplicationRunner {
 
+  private static final Logger log = LoggerFactory.getLogger(RemoteApplicationRunner.class);
+
   public RemoteApplicationRunner(Config config) {
     super(config);
   }
 
-  @Override public void run(StreamGraphBuilder app, Config config) {
-    // TODO: add description of ProcessContext that is going to create a sub-DAG of the {@code graph}
-    // TODO: actually instantiate the tasks and run the job, i.e.
-    // 1. create all input/output/intermediate topics
-    // 2. create the single job configuration
-    // 3. execute JobRunner to submit the single job for the whole graph
-  }
+  /**
+   * Run the {@link StreamGraphBuilder} on the remote cluster
+   * @param app a StreamApplication
+   */
+  @Override
+  public void run(StreamGraphBuilder app) {
+    try {
+      // 1. build stream graph
+      StreamGraph streamGraph = new StreamGraphImpl(this, config);
+      app.init(streamGraph, config);
 
+      // 2. create the physical execution plan
+      ExecutionPlanner planner = new ExecutionPlanner(config);
+      ProcessorGraph processorGraph = planner.plan(streamGraph);
+
+      // 3. submit jobs for remote execution
+      processorGraph.getProcessors().forEach(processor -> {
+          Config processorConfig = processor.generateConfig();
+          log.info("Starting processor {} with config {}", processor.getId(), config);
+          JobRunner runner = new JobRunner(processorConfig);
```

Comment:
This remote runner can start acting weird or most likely, fail if the user has configured "job.factory.class". This seems totally possible considering the number of configs we require from the user :)